chefkoch/recipe.py

```diff
@@ -0,0 +1,771 @@
+            # only abort if the file is missing. Missing keyword is possible.
+            try:
+                forgetfile = entry['file']
+            except KeyError as err:
+                raise ValueError("There is no file given to the file param!")
+                # todo: is it important to have a value error or can I change
+                # it to KeyError?
+            pass
+        except IOError as err:
+            # given filepath does not exist, so stop execution
+            raise ValueError("The filepath to the file param does not exist.")
+            # todo ValueError is catched somewhere. Rename to IOError there!
+
+    def appendValuesFromRange(self, entry):
+        """
+        Appends all values within a range given in the JSON data to Param.values
+        
+        :param entry:       dict with fields start, stop and step.
+        :type entry:        dictionary
+        :raises KeyError: If a field of the entry is missing.
+        """
+        logger.debug("More values are given by a range.")
+        try:
+            i = entry['start']
+            stop = entry['stop']
+            step = entry['step']
+        except KeyError as err:
+            raise KeyError("The start, stop or step field of " + str(entry) +
+                            "are missing.")
+        valid_start = isinstance(i, int) or isinstance(i, float)
+        valid_stop = isinstance(stop, int) or isinstance(stop, float)
+        valid_step = isinstance(step, int) or isinstance(step, float)
+        valid = valid_start and valid_step and valid_stop
+        if not valid:
+            logger.warning("The start, step and stop value of a parameter range" +
+                           " need to by of type int or float, so an empty list" + 
+                           " was appended! Check for correctness!")
+            return
+        # get the direction of the range before using <= or >=
+        if i < stop and step > 0:
+            # add all values within range
+            while i <= stop:
+                logger.debug("Adding value " + str(i))
+                self.values.append(i)
+                i = i + step
+        elif i > stop and step < 0:
+            # add all values within range
+            while i >= stop:
+                logger.debug("Adding value " + str(i))
+                self.values.append(i)
+                i = i + step
+        else:
+            logger.warning("The start " + str(i) + ", stop " + str(stop) + " and step " +
+                           str(step) + " of the range leave an empty list for" +
+                          " this paramter. Please check if this is intended.")
+
+    def appendEntry(self, entry):
+        """
+        Appends a single entry within the JSON data received from the flavour file.
+        
+        :param entry:       file or range or any other value
+        :type entry:        dictionary
+        """
+        try:
+            type = entry['type']
```

Comment:
avoid magic strings in the code --> define a properly named constant at one place in the codebase